software/contrib/consequencer.py

```diff
@@ -5,100 +5,90 @@
 
-A gate and CV sequencer inspired by Grids from Mutable Instruments that contains pre-loaded drum patterns that can be smoothly morphed from one to another. Triggers are sent from outputs 1 - 3, randomized stepped CV patterns are sent from outputs 4 - 6.
-Send a clock to the digital input to start the sequence.
+Consequencer is a gate and stepped CV sequencer inspired by Grids from Mutable Instruments and the randomness created by the Turing Machine.
+Pattern changes and randomness are introduced as a consequence of both manual input and control voltages sent to the analogue input.
+A large number of popular gate patterns are pre-loaded. Stepped CV sequences are automatically generated.
 
-Demo video: https://youtu.be/UwjajP6uiQU
+"""
 
-digital_in: clock in
-analog_in: Mode 1: Adjusts randomness, Mode 2: Selects gate pattern, Mode 3: Selects stepped CV pattern
-
-knob_1: randomness
-knob_2: select pre-loaded drum pattern
-
-button_1: Short Press: toggle randomized hi-hats on / off. Long Press: Play previous CV Pattern
-button_2:
-- Short Press  (<300ms)  : Generate a new random cv pattern for outputs 4 - 6.
-- Medium Press (>300ms)  : Cycle through analogue input modes
-- Long Press   (>3000ms) : Toggle option to send clocks from output 4 on / off
-
-output_1: trigger 1 / Bass Drum
-output_2: trigger 2 / Snare Drum
-output_3: trigger 3 / Hi-Hat
-output_4: randomly generated CV (cycled by pushing button 2)
-output_5: randomly generated CV (cycled by pushing button 2)
-output_6: randomly generated CV (cycled by pushing button 2)
-
-'''
-
-'''
+"""
 Version History
 March-23    decreased maxRandomPatterns to 32 to avoid crashes on some systems
             pattern is now sum of ain and k2
             randomness is now sum of ain and k1
             added garbage collection to avoid memory allocation errors when creating new random sequences
             scroll pattern on display
             minor pattern updates and reshuffled the order
-'''
+
+Jan-24      reduced number of calls to update screen to improve performance for incoming clocks < 7ms
+            added grids patterns as an easter egg
+            added methods to reduce hysteresis on ain - this reduces the number of ain reads too
+            added methods to reduce hysteresis on k1 and k2 - this reduces the number of knob reads too
+            added screen saver to improve performance
+            changed the way the mode is displayed - replaced M1,M2,M3 with Mr,Mp,Mc for easier reading
+            cleaned up some code comments
+            added constants for easier code reading
+"""
 
 # Operating modes for the internal state machine
 MODE_RANDOM = 1
 MODE_PATTERN = 2
 MODE_CV_PATTERN = 3
 
+# Change detection thresholds to trigger a value change
+KNOB_CHANGE_TOLERANCE = 0.999
+AIN_CHANGE_TOLERANCE = 1
+
+# AIN Input Mode display chars
+MODE_DISPLAY_CHARS = ["r", "p", "c"]
+
+# Wake the screen upon detecting input at ain?
+WAKE_SCREEN_ON_AIN_INPUT = False
```

Comment:
This seems to be contrary to the README which states "Example interactions are : ain input"

Reply:
Good spot. I removed ain input as a valid interaction during further testing as it seemed to be more of an interference than a nice feature to have. Updated the doc.

Reply:
Nice